.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+version: 2
+
+references:
+  container_config: &container_config
+    setup_remote_docker:
+      docker_layer_caching: true
+    docker:
+      - image: aetrnty/builder
+        user: builder
+    working_directory: ~/infrastructure
+
+  venv_cache_key: &venv_cache_key v2-venv-cache-{{ checksum "requirements.txt" }}
+  restore_venv: &restore_venv
+    restore_cache:
+      key: *venv_cache_key
+
+  roles_cache_key: &roles_cache_key v1-ansible-roles-{{ checksum "ansible/requirements.yml" }}
+  restore_roles: &restore_roles
+    restore_cache:
+      key: *roles_cache_key
+
+  install_pip: &install_pip
+    run:
+      name: Install PIP dependancies
+      command: pip install -r requirements.txt
+
+  activate_venv: &activate_venv
+    run:
+      name: Activate python virtual environment
+      command: |
+        source ~/infrastructure/venv/ansible/bin/activate
+        echo "source ~/infrastructure/venv/ansible/bin/activate" >> $BASH_ENV
+
+jobs:
+  build:
+    <<: *container_config
+    steps:
+      - checkout
+      - *restore_venv
+      - run:
+          name: Create python virtual environment
+          command: |
+            virtualenv venv/ansible
+      - *activate_venv
+      - *install_pip
+      - save_cache:
+          key: *venv_cache_key
+          paths:
+            - venv/ansible
+      - run:
+          name: Install ansible roles
+          command: |
+            cd ansible && ansible-galaxy install -r requirements.yml -p roles
+      - save_cache:
+          key: *roles_cache_key
+          paths:
+            ansible/roles
+  check:
+    <<: *container_config
+    steps:
+      - checkout
+      - *restore_venv
+      - *activate_venv
+      - run:
+          name: Lint ansible playbooks
+          command: |
+            ansible-lint ansible/setup.yml
+            ansible-lint ansible/monitoring.yml
+            ansible-lint ansible/manage-node.yml
+            ansible-lint ansible/reset-net.yml
+      - run:
+          name: Dry run OpenStack heat template
+          command: |
+            openstack stack create test -e openstack/test/create.yml \
+              -t openstack/ae-environment.yml --enable-rollback --wait --dry-run
+      - run:
+          name: Dry run environments playbooks
+          command: |
+            cd ansible && ansible-playbook -e 'ansible_python_interpreter="/usr/bin/env python"' \
+              --check -i localhost, environments.yml
+
+  deploy:
+    <<: *container_config
+    steps:
+      - checkout
+      - *restore_venv
+      - *activate_venv
+      - *restore_roles
+      # Don't run on UAT until we're sure it won't cause changes
+      - run:
+          name: Setup infrastructure
+          command: |
+            cd ansible && ansible-playbook -e 'ansible_python_interpreter="/usr/bin/env python"' \
+              --skip-tags "uat-env" environments.yml
```

Comment:
I see
https://github.com/aeternity/infrastructure/blob/5308db4e90e8a73ad27162230b3e9097568fb885/ansible/environments.yml#L57
http://manpages.ubuntu.com/manpages/xenial/en/man1/ansible-playbook.1.html
so shall be ok. Though still keep an eye on envs when merging.